libvast/vast/format/ostream_writer.hpp

```diff
@@ -84,27 +85,33 @@ class ostream_writer : public writer {
   /// @returns `ec::print_error` if `printer` fails to generate output,
   ///          otherwise `caf::none`.
   template <class Policy, class Printer>
-  caf::error print(Printer& printer, const table_slice& x,
+  caf::error print(Printer& printer, const table_slice& xs,
                    std::string_view begin_of_line, std::string_view separator,
                    std::string_view end_of_line) {
-    auto at = [&](size_t row, size_t column) {
-      if constexpr (std::is_same_v<Policy, policy::include_field_names>) {
-        return std::pair{x.column_name(column), x.at(row, column)};
-      } else {
-        static_assert(std::is_same_v<Policy, policy::omit_field_names>,
-                      "Unsupported policy: Expected either include_field_names "
-                      "or omit_field_names");
-        return x.at(row, column);
-      }
+    auto print_field = [&](auto& iter, size_t row, size_t column) {
+      auto rep = [&](data_view x) {
+        if constexpr (std::is_same_v<Policy, policy::include_field_names>) {
+          return std::pair{xs.column_name(column), x};
+        } else {
+          static_assert(std::is_same_v<Policy, policy::omit_field_names>,
+                        "Unsupported policy: Expected either "
+                        "include_field_names "
+                        "or omit_field_names");
+          return x;
+        }
```

Comment:
In other parts of the code-base, we use this pattern for static dispatching on the policy:

```cpp
if constexpr (std::is_same_v<Policy, policy::include_field_names>)
  return std::pair{xs.column_name(column), x};
else if constexpr (std::is_same_v<Policy, policy::omit_field_names>)
  return x;
else
  static_assert(detail::always_false_v<Policy>, ...);
```

I kinda like this better, because it makes the policy comparison explicit. It's also more extensible when adding future policies.

Reply:
Noted. I applied only the necessary changes to this function, but left the style as it was otherwise. I'll implement your suggestion.